package/MDAnalysis/core/groups.py

```diff
@@ -3324,6 +3324,66 @@ def write(self, filename=None, file_format=None,
 
         raise ValueError("No writer found for format: {}".format(filename))
 
+    def sort(self, key='ix', keyfunc=None):
+        """Return stably sorted Atomgroup by a key that specifies
+           an attribute of atomgroup.
+
+        Parameters
+        ----------
+        keys: str
+            The name for attribute of Atomgroup. (e.g. "ids", "ix". default="ix".)
+        keyfunc: function
+            A function that returns 1 dimension array, a key for sorting the
+            atomgroup from multiple dimension array that the attribute returns.
+
+        Returns
+        -------
+        output:
+            Sorted atomgroup.
+
+        Example
+        -------
+
+        .. code-block:: python
+
+            >>> import MDAnalysis as mda
+            >>> from MDAnalysisTests.datafiles import PDB_small
+            >>> u = mda.Universe(PDB_small)
+            >>> ag = u.atoms[[3, 2, 1, 0]]
+            >>> ag.ix
+            array([3 2 1 0])
+            >>> ag = ag.sort()
+            >>> ag.ix
+            array([0 1 2 3])
+
+        Note
+        ----
+        This sort is stable as it is implemented by
+        `numpy.argsort(kind='stable')`.
+
+        .. versionadded:: 2.0.0
+        """
+        idx = getattr(self.atoms, key)
+        if len(idx) != len(self.atoms):
+            raise ValueError("The array returned by the attribute '{}' "
+                             "must have a same length as the number of "
+                             "atoms".format(key))
+        if idx.ndim == 1:
+            order = np.argsort(idx, kind='stable')
+        elif idx.ndim > 1:
+            if keyfunc is None:
+                raise NameError("You have to assign the argument 'keyfunc' "
```

Comment:
I think this message needs reversing in order to be more user friendly.

"The {} attribute returns a multidimensional array. In order to sort it, a function returning a 1D array (to be used as the sort key) must be passed to the `keyfunc` argument"

Reply:
Fixed. 

https://github.com/MDAnalysis/mdanalysis/pull/3188/commits/e0bef58681c2fba19490ceb1d964740f60518e2d
https://github.com/MDAnalysis/mdanalysis/pull/3188/commits/338417e2ea1c71d908d22037412a8d7d5c014b72